pandas/tests/indexes/interval/test_interval_new.py

```diff
@@ -314,3 +314,197 @@ def test_contains_method(self):
+        tm.assert_numpy_array_equal(Interval(0, 5, closed='both').covers(idx), np.array([0, 1, 2]))
+        tm.assert_numpy_array_equal(Interval(0, 2, closed='both').covers(idx), np.array([0]))
+        tm.assert_numpy_array_equal(Interval(2, 4, closed='both').covers(idx), np.array([1]))
+
+    def test_interval_overlaps_interval(self):
+
+        # class Interval:
+        #     def overlaps(self, other: Interval) -> bool
+
+        assert Interval(1, 3).overlaps(Interval(1.5, 2.5))
+        assert Interval(1, 3).overlaps(Interval(1, 2))
+        assert Interval(1, 3).overlaps(Interval(2, 3))
+        assert Interval(1, 3).overlaps(Interval(0.5, 2.5))
+        assert Interval(1, 3).overlaps(Interval(1.5, 3.5))
+
+        assert not Interval(1, 3).overlaps(Interval(-1, 1))
+        assert not Interval(1, 3).overlaps(Interval(3, 5))
+
+        # right
+        assert Interval(1, 3, closed='right').overlaps(Interval(1, 3, closed='right'))
+        assert Interval(1, 3, closed='right').overlaps(Interval(1, 3, closed='left'))
+        assert Interval(1, 3, closed='right').overlaps(Interval(1, 3, closed='both'))
+
+        assert not Interval(1, 3, closed='right').overlaps(Interval(-1, 1, closed='right'))
+        assert not Interval(1, 3, closed='right').overlaps(Interval(-1, 1, closed='left'))
+        assert not Interval(1, 3, closed='right').overlaps(Interval(-1, 1, closed='both'))
+
+        assert not Interval(1, 3, closed='right').overlaps(Interval(3, 5, closed='right'))
+        assert Interval(1, 3, closed='right').overlaps(Interval(3, 5, closed='left'))
+        assert Interval(1, 3, closed='right').overlaps(Interval(3, 5, closed='both'))
+
+        # left
+        assert Interval(1, 3, closed='left').overlaps(Interval(1, 3, closed='right'))
+        assert Interval(1, 3, closed='left').overlaps(Interval(1, 3, closed='left'))
+        assert Interval(1, 3, closed='left').overlaps(Interval(1, 3, closed='both'))
+
+        assert not Interval(1, 3, closed='left').overlaps(Interval(-1, 1, closed='right'))
+        assert not Interval(1, 3, closed='left').overlaps(Interval(-1, 1, closed='left'))
+        assert not Interval(1, 3, closed='left').overlaps(Interval(-1, 1, closed='both'))
+
+        assert not Interval(1, 3, closed='left').overlaps(Interval(3, 5, closed='right'))
+        assert Interval(1, 3, closed='left').overlaps(Interval(3, 5, closed='left'))
+        assert Interval(1, 3, closed='left').overlaps(Interval(3, 5, closed='both'))
+
+        # both
+        assert Interval(1, 3, closed='both').overlaps(Interval(1, 3, closed='right'))
+        assert Interval(1, 3, closed='both').overlaps(Interval(1, 3, closed='left'))
+        assert Interval(1, 3, closed='both').overlaps(Interval(1, 3, closed='both'))
+
+        assert Interval(1, 3, closed='both').overlaps(Interval(-1, 1, closed='right'))
+        assert not Interval(1, 3, closed='both').overlaps(Interval(-1, 1, closed='left'))
```

Comment:
btw I would be very happy to have these in a different file, e.g.

``pandas/tests/indexes/interval/test_overlap_covers.py`` or whatever (excluding @jschendel commet that the Interval tests need to go in the scalar tests); we can make a sub-dir there as well I think.